controllers/routing/routing_controller.go

```diff
@@ -75,6 +79,8 @@ func (r *PlatformRoutingReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		errs = append(errs, reconciler(ctx, sourceRes))
 	}
 
+	controllerutil.AddFinalizer(sourceRes, metadata.Finalizers.Routing)
```

Comment:
I think we should add the finalizer first before we ensure the cluster is in the desired state (so before we call any reconciler).

Otherwise, we risk ending in borked state, if we fail on reconcile partially. We wouldn't be able to clean up orphaned resources.

Reply:
[ee4f949](https://github.com/maistra/odh-platform/pull/39/commits/ee4f9496ec4c80b2113500ba3f1dceb8b09131e7)